build/sentinel.go

```diff
@@ -0,0 +1,3 @@
+package build
+
+var SentinelMode string = "false"
```

Comment:
Setting and parsing this seems really clunky. Is this the pattern already in use by Lotus? If so then let's leave it (but it's still clunky!)

In theory the Go compiler could optimize away code that isn't needed for sentinel mode but I think this would have to be a const bool for that to work.

I feel that fx must have better support for compile time switching of modules.

Reply:
The only prior art I saw for this was passing the version via a build flag. @magik6k wants to avoid adding more flags to the `daemon` and suggested a build flag or a go plugin. If, as you've already suggested, I can pull the required bits into a separate binary, then I think this flag can be removed.

Reply:
Unless separate binary, using [build tags](https://www.digitalocean.com/community/tutorials/customizing-go-binaries-with-build-tags) might be cleaner: `go build -tags sentinel`. It also forces to separate everything sentinel-specific to its own files.

Reply:
I've implemented a standalone binary here: https://github.com/filecoin-project/lotus/pull/5693